flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/persistent/PersistentKafkaSource.java

```diff
@@ -225,7 +226,16 @@ public void close() {
 
 	@Override
 	public void restoreState(long[] state) {
-		// we maintain the offsets in Kafka, so nothing to do.
+		if(lastOffsets == null) {
+			LOG.warn("Restore state called before open() has been called");
+			return;
+		}
+		LOG.info("Restoring state to {}", Arrays.toString(state));
+		if(lastOffsets.length != state.length) {
```

Comment:
How about sanity checking before logging that things are going to happen? Usually gives better logging insights...